projects/solar/src/relay_fsm.c

```diff
@@ -2,17 +2,64 @@
 
 #include <stdbool.h>
 
+#include "can_transmit.h"
+#include "can_unpack.h"
+#include "data_store.h"
 #include "event_queue.h"
 #include "fault_handler.h"
 #include "fsm.h"
+#include "gpio.h"
 #include "log.h"
+#include "solar_boards.h"
 #include "solar_config.h"
 #include "solar_events.h"
 #include "status.h"
 
+static SolarMpptCount solar_mppt_count;
+
+static void prv_assert_relay(SoftTimerId timer_id, void *context) {
+  RelayFsmStorage *storage = (RelayFsmStorage *)context;
+  if (storage->isErrCalled == true) {
+    fault_handler_raise_fault(EE_SOLAR_RELAY_OPEN_ERROR, 0);
+    return;
+  }
+
+  int32_t data_value;
+  bool isSet = false;
+
+  data_store_get_is_set(DATA_POINT_CURRENT, &isSet);
+
+  if (!isSet) {
+    if (storage->isSetCounter >= 30) {
```

Comment:
Factor the 30 out into a #define constant with a good name - usually we try not to leave these 'magic numbers' in code without an explanation.

Reply:
done